R/combineSNV_CNA.R

```diff
@@ -0,0 +1,146 @@
+# combineSNV_CNA.R
+
+#' Combine mutation data from CNA and SNV
+#'
+#' \code{combineSNV_CNA} loads the vector of preprocessed dataset from filtered CNA or SNV files,
+#'  creates hash keys corresponding to the input file type, combine the corresponding mutation count
+#'  to each gene and save to a gX format file as output file.
+#'
+#' Details.
+#' @section gX files:
+#'  gX rds files contain data of SNV and CNA combined for each gene as one object, sorted per
+#'  gene.
+#'
+#'
+#' @param fNameCNA A vector of path of fully qualified file names of SNV rds files.
+#' @param fnameSNV A vector of path of fully qualifed filenames of CNA rds files.
+#' @param fgX The fully qualified filename/path of a gX output file, by default is gX.rds.
+#' @param silent Boolean option for writing combining process information to console, FALSE by default.
+#' @param writeLog Boolean option for log results, TRUE by default.
+#'
+#'
+#' @examples
+#' \dontrun{combine(fNameSNV="SNV.rds", fnameCNA="CNA.rds", fgX="combined_data.rds", silent=TRUE, writeLog=FALSE)}
+#'
+#' @export
+combineSNV_CNA <- function(fnameSNV=c(), fnameCNA=c(), fgX="gX.rds", silent=FALSE, writeLog=TRUE) {
+    # parameter validation
+    if (length(fnameSNV) == 0 && length(fnameSNV) == 0) {
+        stop("Input vectors are is empty!")
+    }
+    cR <- character()
+    cR <- c(cR, .checkArgs(fnameSNV,        like = c("FILE_E")))
```

Comment:
I think since they are vectors, you need to check that each file in the vector is like a FILE_E